### ISSUE TITLE
Problem(Argentina): ingested data is not defined

### ISSUE THREAD

**Issue by outerlook** [trufnetwork/adapters#15]

**Comment by outerlook** [comment]
Hey @Ivan-TRUF

https://docs.google.com/spreadsheets/d/1fvYSw_PMtv-CJXS-3riwEzMKIodtwvlojA8VNeGQQCQ/edit?gid=638096968#gid=638096968
has some info about data for a single day (2024-12-16)

I didn't check other days yet.

Some data about it:
- around 54K unique products
- 10M rows, merging all from `productos` on each directory

Is this data expected to be aggregated somehow before deploying to TN? 

I'll soon zip the merged products table and share it here. Uncompressed it's 1.2GB

**Comment by Ivan-TRUF** [comment]
In order to trim down the file size, we could store only the Columns in `all_products` that we will use:
id_producto, productos_descripcion & productos_precio_lista and we can calculate the Average price from there

Does it make sense?

**Comment by outerlook** [comment]
yes, that does. That is the `with_avg_price` result on the link @Ivan-TRUF

**Comment by Ivan-TRUF** [comment]
@outerlook yes, noticed that one. But in step before, we can trim down the rest of columns as well since we won't be using them at all

**Comment by outerlook** [comment]
uncompressed 650MB: https://drive.google.com/file/d/1-RJN6ue-Tu8MLjdx1Yurwl8mblao5FKO/view?usp=drive_link

columns:
```
id_producto,productos_descripcion,productos_precio_lista,date
```

date just to preserve the data I'll need to aggregate with this

is this OK?

**Comment by outerlook** [comment]
note: I'm fixing issues on the `avg_price` about id's with bad format such as `"007,7929e+012"`. The objective here is sending you just to analyze next steps, and not that these files will feed something, correct?

**Comment by Ivan-TRUF** [comment]
@outerlook I'm not sure what you need from me here?

**Comment by outerlook** [comment]
I'm following this step here:

> Please start working on Step 1 first and share the result of 1.b) since 1.c) might change, depending on how the data will look like in 'MASTER productos' table

if this is OK, just a clarification: each id_producto is expected to be a stream on TN, or will we aggregate this further?

**Comment by Ivan-TRUF** [comment]
Ok, context helps - thanks!
@outerlook  Then I have question, I'm trying to check the calculation of some of these producto_id but can't seem to find it in the file here
https://drive.google.com/file/d/1-RJN6ue-Tu8MLjdx1Yurwl8mblao5FKO/view?usp=drive_link
![image](https://github.com/user-attachments/assets/c91ca2e9-7860-45e6-b609-48366428bb22)

**Comment by Ivan-TRUF** [comment]
@outerlook 
Additionally, there seems to be somthing off with calculations > please check product_id 1339
According to numbers from file you shared, it should be 422.105 but in sample file you shared, it is 421.7948718

**Comment by Ivan-TRUF** [comment]
> I'm following this step here:
> 
> > Please start working on Step 1 first and share the result of 1.b) since 1.c) might change, depending on how the data will look like in 'MASTER productos' table
> 
> if this is OK, just a clarification: each id_producto is expected to be a stream on TN, or will we aggregate this further?
@markholdex what are limitations from TN side in regard to having 50k+ data feeds added for ARG gov data?

**Comment by outerlook** [comment]
> I'm trying to check the calculation of some of these producto_id but can't seem to find it in the file here

I fixed now, spreadsheet messed the column type

> Additionally, there seems to be somthing off with calculations > please check product_id 1339
> According to numbers from file you shared, it should be 422.105 but in sample file you shared, it is 421.7948718

I'll check this, thank you

**Comment by Ivan-TRUF** [comment]
@outerlook please check other random samples as well, so we keep everything efficient

**Comment by markholdex** [comment]
@Ivan-TRUF I think there is a problem with the current spreadsheet and the data we want to ingest: 

Mainly the data is just the prices of various products and not prices of Eggs or Dairy or Milk or other subcategory. The problem is that often we don't know what product it is and thus how usable the data is. 

Milka Chocotale with Nuts - 10$
Milka Chocotale with Cookies - 11$
Milka Black Chocotale - 8.99$

If let's say we ingest every single entity in the TN and we now have data streams for different Milka Chocolates from Argentina. Not sure what customer would be interested in querying individual Milka prices and neither Aggregated Milka prices. And while I know what Milka is, I don't know without googling what is `Oblea Opera  92 Gr and 55 gr`. Is it a drink or a food?

What I'm trying to point, is that individual prices don't have that much business value vs the grouped or aggregated prices of these. 

I believe the data should be somehow aggregated, cleansed, and categorized before it is pushed to TN. We might need to create a mapping or leave it to an AI to do it, where individual products and prices are being linked with their respective category and names so we can always know which entity we need to update when the new price arrives. 

We never operated with more than 400 data streams. That's the number we have for US Inflation. Combining 50k+ data streams in real-time every time a refresh is made on the website or a request is done through SDK can actually fail.

1. Is my assumption right around grouping and aggregating?
2. Is it correct if the grouping and categorization would be done outside of TN? So in the end TN ingests only around 100 values.
3. Do you have an example outcome of data that you expect to obtain after we ingest these prices? For example: we expect in the end to be able to query the aggregated price of dairy products. So that we could use them in inflation calculation.

**Comment by outerlook** [comment]
> According to numbers from file you shared, it should be 422.105 but in sample file you shared, it is 421.7948718

hey @Ivan-TRUF, I couldn't reach on the same result as you. I found `421.7948718` again. Do you mind sharing how you did this so I could test simlarly?

I with a notebook again, but also manually extracting roles from a text document

maybe we're talking about different files? we're comparing
- https://drive.google.com/file/d/1-RJN6ue-Tu8MLjdx1Yurwl8mblao5FKO/view?usp=drive_link `with_avg` table
- [full data](https://drive.google.com/file/d/1-RJN6ue-Tu8MLjdx1Yurwl8mblao5FKO/view?usp=drive_link) 

right?

I also tested for `7790250019008`, `7790250019008`, `7790250912811`, `7790265002217`

**Comment by Ivan-TRUF** [comment]
@markholdex 
I would say that all those questions are valid and correct.
The issue/question I see here is what is the end goal? Do we want to build the API for ARG gov for public use or we want to create it for us and potentially for TN users?
Let me book a call tomorrow so we can discuss further

**Comment by Ivan-TRUF** [comment]
@outerlook here is the calculation I did by using data from file from this Drive (https://drive.google.com/file/d/1-RJN6ue-Tu8MLjdx1Yurwl8mblao5FKO/view)
![image](https://github.com/user-attachments/assets/0432aaca-11df-4e10-a9f2-5b82ea3cd196)

**Comment by Ivan-TRUF** [comment]
Probably - not sure why there is only 38 in the file taht is in Google Drive? @outerlook

**Comment by outerlook** [comment]
Trying to debug this: when downloading and accessing via text seems to be 39. Is it office? I'll test with online version of it

**Comment by Ivan-TRUF** [comment]
@outerlook yes, Office and I checked again, it is still 38

**Comment by outerlook** [comment]
hey @Ivan-TRUF, do you mind checking what is the total row number for all products data? 

not sure if this is the case:

> CSV files have no limit of rows you can add to them. Excel won't hold more that the 1 million lines of data if you import a CSV file having more lines.
> 
> Excel will actually ask you whether you want to proceed when importing more than 1 million data rows. It suggests to import the remaining data by using the text import wizard again - you will need to set the appropriate line offset.

source: https://stackoverflow.com/a/23758900/11832970

**Comment by Ivan-TRUF** [comment]
1,048,576 > limitation of Excel, you were right!

**Comment by outerlook** [comment]
they should have made this more explicit about truncation :sweat_smile: thanks for checking!

**Comment by Ivan-TRUF** [comment]
Ok, if you could check few product_ids manually so we can be sure the calculations are working correctly that would be great!

For next steps, please wait for tomorrow and we will share more details

**Comment by markholdex** [comment]
log of the call with the team:

- We decided that data must be categorized first before being ingested in TN.
- @Ivan-TRUF will work on categorizing the current demo sheet
- We decided we need to be warned when new data appears or when errors are happening during spreadsheet processing

**Comment by outerlook** [comment]
@Ivan-TRUF 

- [products descriptions](https://drive.google.com/file/d/14nGr4j8HSDtcE__NzhS3GSQ05VN7Gm53/view?usp=sharing): all products, unique by `id_producto`, ordered by tokenized `productos_descripcion` just to be easier to categorize

Note: I couldn't share directly with google sheets: it's not preserving the `id_producto` as a string on import. `|` is the separator because there are commas in some lines.

see this line:
```
07,79022e+012|Harina Graciela Real 000 1Kg
```

I guess `07,79022e+012` should be preserved as a string, as that's how it comes from SEPA documents, not some rounded number.

**Comment by outerlook** [comment]
I forgot one point on this: this is only based on a single day data (2024-12-16).

Should I try to aggregate all available data to get all products, or does this suffice for now?

**Comment by Ivan-TRUF** [comment]
Let's see how it works for now, and we can easily add on top

**Comment by outerlook** [comment]
done as already defined; we are aggregating into categories mapped by a CSV
